### PR TITLE
Stream LLM Messages and Descriptive Agent Action Logging

### LINKED ISSUE #4: possibility to stream responses from agent
to recreate the issue:

ask for "Tell me 10 jokes"
expectation: each character to be streamed to the client

usecase: I want the user to be informed of what is happening on the backend(intermediate steps etc)

### PR DESCRIPTION
**Pull request by arihanv**
I made changes that should resolve #4.

### CODE REVIEW

**Comment by arihanv** [comment]
I also removed several unused state hooks.

**Comment by tedspare** [comment]
Looks much cleaner and more readable already! Will review more fully after testing it locally

File: templates/agent/src/agents/basic.ts [hunk 1]
```diff
@@ -49,7 +42,15 @@ export async function basicAgent({input}) {
 		agentArgs: {prefix},
 		agentType: 'openai-functions',
 		returnIntermediateSteps: env.NODE_ENV === 'development',
-		verbose: env.NODE_ENV === 'development'
+		verbose: env.NODE_ENV === 'development',
+		callbacks: env.NODE_ENV === 'development' && [
+			{
+				async handleAgentAction(action, runId, parentRunId, tags) {
```
Comment (tedspare) [inline templates/agent/src/agents/basic.ts:48]: Can remove unused params

File: templates/agent/src/components/ChatBox.tsx [hunk 2]
```diff
@@ -71,22 +78,25 @@ export default function ChatBox({refetch}: Props) {
...
-					...prevData,
-					`\n<code>${JSON.parse(objects[objectIndex])[0].generations[0][0]}</code>\n`
-				])
-			} else if (inObject)
-				setObjects(prevObjects => {
-					const newObjects = [...prevObjects]
-					newObjects[objectIndex] += text
-					return newObjects
+			// Check if an action call is made and get the according message object
+			let [f, m] = getMessage(text)
```
Comment (tedspare) [inline templates/agent/src/components/ChatBox.tsx:82]: Could use more descriptive variable names here

**Review by tedspare — APPROVED** [review]
LGTM!

Everything looks good in some basic local tests:

image

**Pull request merged**